tests/cpp/test_allocation_order_inference.cpp

```diff
@@ -124,82 +135,9 @@ TEST_F(AllocationOrderInferenceTest, BinaryOpPropagation) {
         tv1->axis(1), tv1->axis(0), tv1->axis(2), tv1->axis(3)};
     tv1->setAllocationDomain(tv1_format, true);
 
-    const auto inferred_layout =
-        preseg_passes::inferenceAllocationOrder(&fusion);
-    EXPECT_THAT(inferred_layout.at(tv2), ElementsAre(1, 0, 2, 3));
-    EXPECT_THAT(inferred_layout.at(tv3), ElementsAre(1, 0, 2, 3));
-  }
-  {
-    auto fusion_ptr = std::make_unique<Fusion>();
-    Fusion& fusion = *fusion_ptr.get();
-    FusionGuard fg(&fusion);
-
-    // Testing propagation between two tensors
-    // tv0 and tv1 has the same number of non-broadcast iter domains, so lhs
-    // operand would propagate its allocation order.
-    auto tv0 = makeSymbolicTensor({-1, -1, 1, 1});
-    fusion.addInput(tv0);
-    auto tv1 = makeSymbolicTensor({-1, -1, 1, 1});
-    fusion.addInput(tv1);
-    // tv2 should have allocation order from tv0
-    auto tv2 = add(tv0, tv1);
-    fusion.addOutput(tv2);
-    // tv3 should have allocation order from tv1
-    auto tv3 = add(tv1, tv0);
-    fusion.addOutput(tv3);
-
-    std::vector<IterDomain*> tv0_format = {
-        tv0->axis(0), tv0->axis(2), tv0->axis(1), tv0->axis(3)};
-    tv0->setAllocationDomain(tv0_format, true);
-    std::vector<IterDomain*> tv1_format = {
-        tv1->axis(1), tv1->axis(0), tv1->axis(2), tv1->axis(3)};
-    tv1->setAllocationDomain(tv1_format, true);
-
-    const auto inferred_layout =
-        preseg_passes::inferenceAllocationOrder(&fusion);
-    EXPECT_THAT(inferred_layout.at(tv2), ElementsAre(0, 2, 1, 3));
-    EXPECT_THAT(inferred_layout.at(tv3), ElementsAre(1, 0, 2, 3));
-  }
-  {
-    auto fusion_ptr = std::make_unique<Fusion>();
-    Fusion& fusion = *fusion_ptr.get();
-    FusionGuard fg(&fusion);
-
-    // Testing propagation between two tensors
-    // tv0 and tv1 has the same number of non-broadcast iter domains, so lhs
-    // operand would propagate its allocation order.
-    auto tv0 = makeSymbolicTensor({-1, -1, 1, 1});
-    fusion.addInput(tv0);
-    auto tv1 = makeSymbolicTensor({-1, -1, 1, 1});
-    fusion.addInput(tv1);
-    // tv2 should have allocation order from tv0
-    auto tv2 = add(tv0, tv1);
-    fusion.addOutput(tv2);
-
-    // reshape propagation is not supported yet
-    auto tv3 = reshape(
```

Comment:
This test has a reshape. Before concretization, it triggers an assert in ComputeAtMap constructor. We are removing this test for that.

Note to myself, the error message is bad. should we check for symbolic dimensions as a hint for Fusion before concretization and throw a meaningful error message instead of assert? cc'ing @jacobhinkle

Reply:
I think it might be best to just manually concretize the fusion before running the pass in this test, mimicking what would happen in a real run. To do that you can use this pattern:
https://github.com/NVIDIA/Fuser/blob/4e8356915fc6ffa823cf97fed4249cc00f317fdf/tests/cpp/test_dynamic_transform.cpp#L71-L72

Reply:
Got ya. I'll give this a try. (note that the existing test might still need to be updated just so that we can have a dominating tensor)